File: 04 Kontrollstrukturen/Go/Aufgabe4.go
package main

import "fmt"

var (
	eingabe string
	langerName string
	fehler bool = false
)

func main() {
	fmt.Print("Bitte gib die Kurzform des Wochentags ein: ")
	fmt.Scan(&eingabe)

	switch eingabe {
		case "mo":
			langerName = "Montag"
		case "di":
			langerName = "Dienstag"
		case "mi":
			langerName = "Mittwoch"
		case "do":
			langerName = "Donnerstag"
		case "fr":
			langerName = "Freitag"
		case "sa":
			langerName = "Samstag"
		case "so":
			langerName = "Sonntag"
		default:
			fehler = true
	}

	if fehler {
		fmt.Println("Es gab einen Fehler bei der Eingabe!")
	} else {
		fmt.Println("Der lange Name des Wochentags ist", langerName, ".");
	}
}

File: 04 Kontrollstrukturen/Pascal/Aufgabe4.pas
Program Aufgabe4;

{$codepage utf8}

Var Eingabe: String;
    LangerName: String;
    Fehler: Boolean = False;

Begin
    Write('Bitte gib die Kurzform des Wochentags ein: ');
    Read(Eingabe);

    Case Eingabe Of
        'mo': LangerName := 'Montag';
        'di': LangerName := 'Dienstag';
        'mi': LangerName := 'Mittwoch';
        'do': LangerName := 'Donnerstag';
        'fr': LangerName := 'Freitag';
        'sa': LangerName := 'Samstag';
        'so': LangerName := 'Sonntag';
    Else
        Fehler := True;
    End;

    If Fehler Then
        WriteLn('Es gab einen Fehler bei der Eingabe!')
    Else
        WriteLn('Der lange Name des Wochentags ist ', LangerName, '.');
End.